internal/provider/datasources/block.go

```diff
@@ -0,0 +1,201 @@
+package datasources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &blockDataSource{}
+	_ datasource.DataSourceWithConfigure = &blockDataSource{}
+)
+
+// blockDataSource is the data source implementation.
+type blockDataSource struct {
+	client api.PrefectClient
+}
+
+// BlockDataSourceModel defines the Terraform data source model.
+type BlockDataSourceModel struct {
+	ID          customtypes.UUIDValue      `tfsdk:"id"`
+	Created     customtypes.TimestampValue `tfsdk:"created"`
+	Updated     customtypes.TimestampValue `tfsdk:"updated"`
+	AccountID   customtypes.UUIDValue      `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue      `tfsdk:"workspace_id"`
+
+	Name          types.String          `tfsdk:"name"`
+	Data          jsontypes.Normalized  `tfsdk:"data"`
+	BlockSchemaID customtypes.UUIDValue `tfsdk:"block_schema_id"`
+	BlockTypeID   customtypes.UUIDValue `tfsdk:"block_type_id"`
+	BlockTypeName types.String          `tfsdk:"block_type_name"`
```

Comment:
im wondering if we just omit these.  

my thought process was - we include attributes on the datasource model if we use them to query OR if they are part of the response + we anticipate a user needing to use them <somewhere>

for query attributes, i think `id` and possibly `name` are valid

for response attributes - functionally, i was thinking that someone only really needs to pull down an existing block if they need (1) the id, so that they can bind it to a `prefect_block_access` resource, or (2) the data itself, so they can pass it into a downstream TF resource

Reply:
Sounds good to me, updated in d6b60ed.